Source/WebCore/Modules/async-clipboard/ClipboardImageReader.cpp

```diff
@@ -27,15 +27,16 @@
 #include "ClipboardImageReader.h"
 
 #include "Document.h"
-#include "NotImplemented.h"
+#include "SharedBuffer.h"
 
 namespace WebCore {
 
 #if !PLATFORM(COCOA)
 
-void ClipboardImageReader::readBuffer(const String&, const String&, Ref<SharedBuffer>&&)
+void ClipboardImageReader::readBuffer(const String&, const String&, Ref<SharedBuffer>&& buffer)
 {
-    notImplemented();
+    if (m_mimeType == "image/png"_s)
+        m_result = Blob::create(m_document.get(), buffer->extractData(), m_mimeType);
```

Comment:
Do I understand correctly that this would also fix [bug #218519](https://bugs.webkit.org/show_bug.cgi?id=218519)?

Reply:
I don't think so